frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -100,6 +102,13 @@ function ItemPicker<TId>({
   const isPickingNotCollection = models.some(model => model !== "collection");
 
   const openCollection = collectionsById[openCollectionId];
+  const isOpenCollectionInPersonalCollection = openCollection?.is_personal;
+  const hideItems =
+    showOnlyPersonalCollections && !isOpenCollectionInPersonalCollection;
+
+  useEffect(() => {
```

Comment:
Why this is needed? The previous solution with calling `onOpenCollectionChange` directly was more straightforward. If there is a reason, can we avoid using `useEffect`?

Reply:
Yeah it's just that `setOpenCollectionId` is passed to many places. Previously `onOpenCollectionChange` was only called when calculating breadcrumbs but not on the collection change.

> If there is a reason, can we avoid using `useEffect`?

Yeah, I think we could encapsulate `setOpenCollectionId`, by wrapping that in a new function that would call both `setOpenCollectionId` and `onOpenCollectionChange` at the same time.

Reply:
Hmm, not sure why I used `useEffect` it seems `setOpenCollectionId` is only called in 2 places. I'll inline `onOpenCollectionChange` back then.

Reply:
Fixed in 6d7c8527f88c6856f0d6db5a53cd10300d2eb459